File: app/src/main/java/com/scp/leagueofquiz/api/database/champion/ChampionDao.java
package com.scp.leagueofquiz.api.database.champion;

import androidx.room.Dao;
import androidx.room.Insert;
import androidx.room.Query;
import com.google.common.util.concurrent.ListenableFuture;
import java.util.List;

@Dao
public interface ChampionDao {
  @Insert
  void insertAll(List<Champion> champions);

  @Query("SELECT * FROM champion")
  ListenableFuture<List<Champion>> findAll();

  @Query("SELECT * FROM champion WHERE name NOT IN (:names) ORDER BY random() LIMIT :quantity")
  ListenableFuture<List<Champion>> findRandomChampsExcept(List<String> names, int quantity);
}

File: app/src/main/java/com/scp/leagueofquiz/api/database/champion/ChampionDao.kt
package com.scp.leagueofquiz.api.database.champion

import androidx.room.Dao
import androidx.room.Insert
import androidx.room.Query
import com.scp.leagueofquiz.api.database.champion.jsonClasses.Spell

@Dao
interface ChampionDao {
    @Insert
    suspend fun insertAll(champions: List<Champion?>?)

    @Query("SELECT * FROM champion")
    suspend fun findAll(): List<Champion>

    @Query("SELECT * FROM champion WHERE name NOT IN (:names) ORDER BY random() LIMIT :quantity")
    suspend fun findRandomChampsExcept(names: List<String>, quantity: Int): List<Champion>

    @Query("SELECT * FROM champion WHERE name NOT IN (:names) ORDER BY random() LIMIT 1")
    suspend fun findRandomChampionExcept(names: List<String>): Champion

    @Query("SELECT * FROM champion ORDER BY random() LIMIT 1")
    suspend fun findRandomAbility(): Champion
}